lib/ansible/modules/database/aerospike/aerospike_migrations.py

```diff
@@ -0,0 +1,535 @@
+                "' output."
+            )
+        except TypeError:
+            self.module.fail_json(
+                msg="namespace stat returned was not numerical"
+            )
+        if namespace_tx != 0 or namespace_rx != 0:
+            return True
+        if namespace_tx == 0 and namespace_rx == 0:
+            return False
+        self.module.fail_json(
+            msg="Unimplemented path in namespace_has_migs"
+        )
+
+    # just calls namespace_has_migs and if any namespace has migs returns true
+    def _node_has_migs(self, node=None):
+        migs = 0
+        self._update_cluster_namespace_list()
+        for namespace in self._namespaces:
+            if self._namespace_has_migs(namespace, node) is True:
+                migs += 1
+        if migs == 0:
+            return False
+        return True
+
+    # create a dictionary to store what each node
+    # returns the cluster key as. we should end up with only 1 dict key,
+    # with the key being the cluster key.
+    def _cluster_key_consistent(self):
+        cluster_keys = {}
+        for node in self._nodes:
+            cluster_key = self._cluster_statistics[node][
+                'cluster_key']
+            if cluster_key not in cluster_keys:
+                cluster_keys[cluster_key] = 1
+            else:
+                cluster_keys[cluster_key] += 1
+        if len(cluster_keys.keys()) is 1 and \
+                self._start_cluster_key in cluster_keys:
+            return True
+        return False
+
+    # ensure all nodes have 'migrate_allowed' in their stats output
+    def _cluster_migrates_allowed(self):
+        for node in self._nodes:
+            node_stats = self._info_cmd_helper('statistics', node)
+            allowed = node_stats['migrate_allowed']
+            if allowed == "false":
+                return False
+        return True
+
+    # calls node_has_migs for each node
+    def _cluster_has_migs(self):
+        migs = 0
+        for node in self._nodes:
+            if self._node_has_migs(node) is True:
+                migs += 1
+        if migs == 0:
+            return False
+        return True
+
+    # calls either local_node_has_migs or cluster_has_migs.
+    def _has_migs(self, local):
+        if local is True:
+            return self._local_node_has_migs()
+        return self._cluster_has_migs()
+
+    def _local_node_has_migs(self):
+        return self._node_has_migs(None)
+
+    # ensures all nodes in the cluster are returning the min cluster size
+    # in their statistics output
+    def _is_min_cluster_size(self):
+        sizes = set()
+        for node in self._cluster_statistics:
+            sizes.add(int(self._cluster_statistics[node]['cluster_size']))
+
+        if (len(sizes)) > 1:  # if we are getting more than 1 size, lets say no
+            return False
+        if (min(sizes)) >= self.module.params['min_cluster_size']:
+            return True
+        return False
+
+    '''Added 4.3:
```

Comment:
If this docstring is for the `_cluster_stable()` method, it must be *inside* that method.